src/outagesim/backup_reliability.jl

```diff
@@ -396,59 +655,62 @@ function survival_with_battery(net_critical_load::Vector, starting_batt_soc_kwh:
             survival = ones(M, N)
             h = mod(t + d - 2, t_max) + 1 #determines index accounting for looping around year
             
-            excess_generation = gen_prod .- net_critical_load[h]
-            max_net_generation = maximum_generation .- net_critical_load[h]
+            excess_generation_kw = gen_prod .- net_critical_loads_kw[h]
+            max_net_generation = maximum_generation .- net_critical_loads_kw[h]
 
             #System fails if net generation is always negative (cannot meet load)
             survival[max_net_generation .< 0, ] .= 0
 
             #Update probabilities to account for generator failures
-            # time_vals["generator_shift"] += @elapsed account_for_generator_failures!(gen_battery_prob_matrix, generator_markov_matrix)
             gen_battery_prob_matrix *= generator_markov_matrix 
-            # account_for_generator_failures!(gen_battery_prob_matrix, generator_markov_matrix)
 
             #Update survival probabilities
-            survival_chance = gen_battery_prob_matrix .* survival
+            #Elementwise, probability that i gens is available * probability of surviving if i gens is available
+            survival_chance = gen_battery_prob_matrix .* survival 
 
             #If marginal survival is false then remove probabilities which did not meet load
             if marginal_survival == false
-                gen_battery_prob_matrix = gen_battery_prob_matrix .* survival
+                gen_battery_prob_matrix = survival_chance
             end
-            #update expected lost load for given outage start time and outage duration
+
             survival_probability_matrix[t, d] = sum(survival_chance)
             #Update generation battery probability matrix to account for battery shifting
-            shift_gen_battery_prob_matrix!(gen_battery_prob_matrix, battery_bin_shift(excess_generation, bin_size, batt_kw, batt_charge_efficiency, batt_discharge_efficiency))
+            shift_gen_battery_prob_matrix!(gen_battery_prob_matrix, battery_bin_shift(excess_generation_kw, bin_size, battery_size_kw, battery_charge_efficiency, battery_discharge_efficiency))
         end
     end
     return survival_probability_matrix
 end
 
 """
-    backup_reliability_inputs(d::Dict, p::REoptInputs; r::Dict)::Dict
+    backup_reliability_reopt_inputs(;d::Dict, p::REoptInputs, r::Dict)::Dict
```

Comment:
Why not just call backup_reliability_inputs in backup_reliability_reopt_inputs instead of duplicating code? 
Or, even better, stop backup_reliability_reopt_inputs before it gets to the part that's a responsibility of backup_reliability_inputs. Then for the two backup_reliability functions, the one with 3 arg can just call backup_reliability_reopt_inputs and pass what is returned to the 1 arg backup_reliability function.